lib/Core/Searcher.cpp

```diff
@@ -273,32 +321,129 @@ RandomPathSearcher::RandomPathSearcher(Executor &_executor)
 RandomPathSearcher::~RandomPathSearcher() {
 }
 
-ExecutionState &RandomPathSearcher::selectState() {
+ExecutionState &RandomPathSearcher::selectStateIgnore() {
   unsigned flips=0, bits=0;
-  PTree::Node *n = executor.processTree->root;
-  
+  PTree::Node* n = executor.processTree->root;
+  BacktrackHelper btHelper;
+
+  while (!n->data) {
+    BacktrackHelper::direction avail_dir = btHelper.avail_dir();
+    switch(avail_dir){
+      case BacktrackHelper::RIGHT:
+        if (n->right) {
+          btHelper.go(BacktrackHelper::RIGHT);
+          n = n->right;
+        } else {
+          btHelper.back();
+          n = n->parent;
+        }
+        break;
+      case BacktrackHelper::LEFT:
+        if (n->left){
+          btHelper.go(BacktrackHelper::LEFT);
+          n = n->left;
+        } else {
+          btHelper.back();
+          n = n->parent;
+        }
+        break;
+      case BacktrackHelper::BOTH:
+        if (!n->left){
+          btHelper.go(BacktrackHelper::RIGHT);
+          n = n->right;
+        } else if (!n->right){
+          btHelper.go(BacktrackHelper::LEFT);
+          n = n->left;
+        } else {
+          if (bits==0) {
+            flips = theRNG.getInt32();
+            bits = 32;
+          }
+          --bits;
+          if (flips&(1<<bits)){
+            btHelper.go(BacktrackHelper::LEFT);
+            n = n->left;
+          } else {  
+            btHelper.go(BacktrackHelper::RIGHT);
+            n = n->right;
+          }
+        }
+        break;
+      case BacktrackHelper::NONE:
+        btHelper.back();
+        n = n->parent;
+    }
+    if (n->data){
+      if (ignoreStates.find(n->data) != ignoreStates.end()){
+        btHelper.back();
+        n = n->parent;
+      }
+    }
+  }
+  return *n->data;
+}
+ExecutionState &RandomPathSearcher::selectStateStandard() {
+  unsigned flips=0, bits=0;
+  PTree::Node* n = executor.processTree->root;
+  BacktrackHelper btHelper;
```

Comment:
This variable is unused.  Also, I would prefer to just rename the old `selectState` to `selectStateStandard` w/o reformatting it.